File: Matrix Audio/matrix.audio.client/src/app/models/order.ts
import { Address } from "./address";
import { Entity } from "./entity";
import { PaymentInfo } from "./payment-info";
import { Promo } from "./promo";

export interface Order extends Entity {
    userId: string;
    orderType: string;
    orderStatus: string;
    orderNumber: string;
    currency: string;
    description?: string;
    subtotal: number;
    total: number;
    dateCreated: Date;
    dateUpdated: Date;
    items: OrderItem[];
    billingAddress?: Address;
    paymentInfo?: PaymentInfo;
    promo?: Promo;
}

export interface OrderItem extends Entity {
    orderId: string;
    productId: string;
    productName: string;
    quantity: number;
    unitPrice: number;
    totalPrice: number;
    totalAfterDiscount: number;
    discount: number;
}

export interface OrderViewModel {
    orderId: string;
    userId: string;
    orderNumber: string;
    orderStatus: string;
    productName: string;
    subtotal: number;
    total: number;
    dateCreated: Date;
}

File: Matrix Audio/Matrix.Audio.Models/Order.cs
namespace Matrix.Audio.Models;
public class Order : Entity
{
    public required string UserId { get; set; }
    public required string OrderType { get; set; }
    public required string OrderStatus { get; set; }
    public required string OrderNumber { get; set; }
    public required string Currency { get; set; } = "USD";
    public string? Description { get; set; }
    public decimal Subtotal { get; set; }
    public decimal Total { get; set; }

    public DateTime DateCreated { get; set; } = DateTime.UtcNow;
    public DateTime DateUpdated { get; set; }

    public List<OrderItem> Items { get; set; } = [];
    public Address BillingAddress { get; set; } = new();
    public PaymentInfo PaymentInfo { get; set; } = new();
    public Promo? Promo { get; set; }
}
